core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -610,13 +610,14 @@ open class TileGroup(var tileInfo: TileInfo, var tileSetStrings:TileSetStrings)
     }
 
     private fun updateTerrainFeatureImage() {
-        if (tileInfo.terrainFeature != terrainFeature) {
-            terrainFeature = tileInfo.terrainFeature
+        if (tileInfo.terrainFeatures != terrainFeatures) {
+            terrainFeatures.clear()
+            terrainFeatures.addAll(tileInfo.terrainFeatures)
             if (terrainFeatureOverlayImage != null) terrainFeatureOverlayImage!!.remove()
             terrainFeatureOverlayImage = null
 
-            if (terrainFeature != null) {
-                val terrainFeatureOverlayLocation = tileSetStrings.getTerrainFeatureOverlay(terrainFeature!!)
+            if (terrainFeatures.isNotEmpty()) {
+                val terrainFeatureOverlayLocation = tileSetStrings.getTerrainFeatureOverlay(terrainFeatures.joinToString(separator = "+"))
```

Comment:
TileSetStrings is optimised for minimal ram usage due to duplicate strings - and this totally creates many duplicates, please change the function in tileSetStrings to add terrain features 'gradually', I think you'll understand once you see the class

Reply:
done